Processing Los Angeles Kings at Anaheim Ducks, ID: 388e9de27b2cf8d2f71146601e83b9c9
Processing Colorado Avalanche at San Jose Sharks, ID: 787d6a9121a568436fbfd18aa16f1a24
Processing Tampa Bay Lightning at Toronto Maple Leafs, ID: 807805467381c6b6ef95ee1de0959cbe
Player: Brayden Point
  Bookmaker: BetMGM, Over/Under: Over, Price: 1.69, Points: 2.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
  Bookmaker: BetMGM, Over/Under: Under, Price: 2.05, Points: 2.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
Player: Nikita Kucherov
  Bookmaker: BetMGM, Over/Under: Over, Price: 1.71, Points: 3.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
  Bookmaker: BetMGM, Over/Under: Under, Price: 2.0, Points: 3.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
Player: Victor Hedman
  Bookmaker: BetMGM, Over/Under: Over, Price: 2.05, Points: 2.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
  Bookmaker: BetMGM, Over/Under: Under, Price: 1.71, Points: 2.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
Player: Max Domi
  Bookmaker: BetMGM, Over/Under: Over, Price: 1.74, Points: 1.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
  Bookmaker: BetMGM, Over/Under: Under, Price: 2.0, Points: 1.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
Player: Oliver Ekman-Larsson
  Bookmaker: BetMGM, Over/Under: Over, Price: 1.5, Points: 1.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
  Bookmaker: BetMGM, Over/Under: Under, Price: 2.45, Points: 1.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
Player: Jake Guentzel
  Bookmaker: BetMGM, Over/Under: Over, Price: 1.62, Points: 2.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
  Bookmaker: BetMGM, Over/Under: Under, Price: 2.2, Points: 2.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
Player: Nicholas Paul
  Bookmaker: BetMGM, Over/Under: Over, Price: 1.61, Points: 1.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
  Bookmaker: BetMGM, Over/Under: Under, Price: 2.25, Points: 1.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
Player: Morgan Rielly
  Bookmaker: BetMGM, Over/Under: Over, Price: 1.5, Points: 1.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
  Bookmaker: BetMGM, Over/Under: Under, Price: 2.45, Points: 1.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
Player: Matthew Knies
  Bookmaker: BetMGM, Over/Under: Over, Price: 1.61, Points: 1.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
  Bookmaker: BetMGM, Over/Under: Under, Price: 2.25, Points: 1.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
Player: Brandon Hagel
  Bookmaker: BetMGM, Over/Under: Over, Price: 2.45, Points: 2.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
  Bookmaker: BetMGM, Over/Under: Under, Price: 1.5, Points: 2.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
Player: William Nylander
  Bookmaker: BetMGM, Over/Under: Over, Price: 1.87, Points: 3.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
  Bookmaker: BetMGM, Over/Under: Under, Price: 1.83, Points: 3.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
Player: Auston Matthews
  Bookmaker: BetMGM, Over/Under: Over, Price: 1.83, Points: 4.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
  Bookmaker: BetMGM, Over/Under: Under, Price: 1.91, Points: 4.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
Player: Mitch Marner
  Bookmaker: BetMGM, Over/Under: Over, Price: 2.15, Points: 2.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
  Bookmaker: BetMGM, Over/Under: Under, Price: 1.62, Points: 2.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
Player: Bobby McMann
  Bookmaker: BetMGM, Over/Under: Over, Price: 1.5, Points: 1.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
  Bookmaker: BetMGM, Over/Under: Under, Price: 2.45, Points: 1.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
Player: John Tavares
  Bookmaker: BetMGM, Over/Under: Over, Price: 1.54, Points: 2.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
  Bookmaker: BetMGM, Over/Under: Under, Price: 2.35, Points: 2.5, Home Team: Toronto Maple Leafs, Away Team: Tampa Bay Lightning
Used requests: 80
Remaining requests: 420
No valid game log data available for player ID: 8484144 before date 2024-10-11 for season 20222023.
No valid game log data available for player ID: 8484144 before date 2024-10-11 for season 20222023.
No valid game log data available for player ID: 8484166 before date 2024-10-15 for season 20222023.
Adam Fantilli has less than 60 GP.
No valid game log data available for player ID: 8484166 before date 2024-10-15 for season 20222023.
Adam Fantilli has less than 60 GP.
No valid game log data available for player ID: 8484227 before date 2024-10-15 for season 20232024.
No valid game log data available for player ID: 8484227 before date 2024-10-15 for season 20222023.
Will Smith has less than 60 GP.
Traceback (most recent call last):
  File "/Users/alexmann/Documents/Code_Learning/nhl_fromscratch/player_api.py", line 260, in <module>
    fetch_and_store_player_data()
  File "/Users/alexmann/Documents/Code_Learning/nhl_fromscratch/player_api.py", line 194, in fetch_and_store_player_data
    player_id = get_player_id(home_team, away_team, player_name)
  File "/Users/alexmann/Documents/Code_Learning/nhl_fromscratch/player_api.py", line 80, in get_player_id
    player_id = search_player_in_team(team1_id)
  File "/Users/alexmann/Documents/Code_Learning/nhl_fromscratch/player_api.py", line 59, in search_player_in_team
    response = requests.get(f"{base_url}/roster/{team_id}/current")
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/requests/api.py", line 73, in get
    return request("get", url, params=params, **kwargs)
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/requests/adapters.py", line 667, in send
    resp = conn.urlopen(
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/urllib3/connectionpool.py", line 789, in urlopen
    response = self._make_request(
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/urllib3/connectionpool.py", line 466, in _make_request
    self._validate_conn(conn)
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/urllib3/connectionpool.py", line 1095, in _validate_conn
    conn.connect()
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/urllib3/connection.py", line 693, in connect
    self.sock = sock = self._new_conn()
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/urllib3/connection.py", line 199, in _new_conn
    sock = connection.create_connection(
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/urllib3/util/connection.py", line 73, in create_connection
    sock.connect(sa)
KeyboardInterrupt
Traceback (most recent call last):
  File "/Users/alexmann/Documents/Code_Learning/nhl_fromscratch/main.py", line 85, in <module>
    main()
  File "/Users/alexmann/Documents/Code_Learning/nhl_fromscratch/main.py", line 55, in main
    subprocess.run(['python3', os.path.join(script_dir, 'player_api.py')])
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/subprocess.py", line 507, in run
    stdout, stderr = process.communicate(input, timeout=timeout)
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/subprocess.py", line 1126, in communicate
    self.wait()
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/subprocess.py", line 1189, in wait
    return self._wait(timeout=timeout)
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/subprocess.py", line 1917, in _wait
    (pid, sts) = self._try_wait(0)
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/subprocess.py", line 1875, in _try_wait
    (pid, sts) = os.waitpid(self.pid, wait_flags)
KeyboardInterrupt
Odds for today are already in the database.
No valid game log data available for player ID: 8484144 before date 2024-10-11 for season 20222023.
No valid game log data available for player ID: 8484144 before date 2024-10-11 for season 20222023.
No valid game log data available for player ID: 8484166 before date 2024-10-15 for season 20222023.
Adam Fantilli has less than 60 GP.
No valid game log data available for player ID: 8484166 before date 2024-10-15 for season 20222023.
Adam Fantilli has less than 60 GP.
No valid game log data available for player ID: 8484227 before date 2024-10-15 for season 20232024.
No valid game log data available for player ID: 8484227 before date 2024-10-15 for season 20222023.
Will Smith has less than 60 GP.
No valid game log data available for player ID: 8484227 before date 2024-10-15 for season 20232024.
No valid game log data available for player ID: 8484227 before date 2024-10-15 for season 20222023.
Will Smith has less than 60 GP.
No valid game log data available for player ID: 8482702 before date 2024-10-15 for season 20222023.
Logan Stankoven has less than 60 GP.
No valid game log data available for player ID: 8482702 before date 2024-10-15 for season 20222023.
Logan Stankoven has less than 60 GP.
No valid game log data available for player ID: 8484180 before date 2024-10-15 for season 20232024.
No valid game log data available for player ID: 8484180 before date 2024-10-15 for season 20222023.
Samuel Honzek has less than 60 GP.
No valid game log data available for player ID: 8484180 before date 2024-10-15 for season 20232024.
No valid game log data available for player ID: 8484180 before date 2024-10-15 for season 20222023.
Samuel Honzek has less than 60 GP.
No valid game log data available for player ID: 8484387 before date 2024-10-15 for season 20232024.
No valid game log data available for player ID: 8484387 before date 2024-10-15 for season 20222023.
Matvei Michkov has less than 60 GP.
No valid game log data available for player ID: 8484387 before date 2024-10-15 for season 20232024.
No valid game log data available for player ID: 8484387 before date 2024-10-15 for season 20222023.
Matvei Michkov has less than 60 GP.
Traceback (most recent call last):
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/urllib3/connectionpool.py", line 789, in urlopen
    response = self._make_request(
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/urllib3/connectionpool.py", line 536, in _make_request
    response = conn.getresponse()
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/urllib3/connection.py", line 507, in getresponse
    httplib_response = super().getresponse()
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/http/client.py", line 1377, in getresponse
    response.begin()
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/http/client.py", line 320, in begin
    version, status, reason = self._read_status()
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/http/client.py", line 281, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/socket.py", line 704, in readinto
    return self._sock.recv_into(b)
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/ssl.py", line 1242, in recv_into
    return self.read(nbytes, buffer)
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/ssl.py", line 1100, in read
    return self._sslobj.read(len, buffer)
ConnectionResetError: [Errno 54] Connection reset by peer

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/requests/adapters.py", line 667, in send
    resp = conn.urlopen(
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/urllib3/connectionpool.py", line 843, in urlopen
    retries = retries.increment(
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/urllib3/util/retry.py", line 474, in increment
    raise reraise(type(error), error, _stacktrace)
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/urllib3/util/util.py", line 38, in reraise
    raise value.with_traceback(tb)
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/urllib3/connectionpool.py", line 789, in urlopen
    response = self._make_request(
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/urllib3/connectionpool.py", line 536, in _make_request
    response = conn.getresponse()
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/urllib3/connection.py", line 507, in getresponse
    httplib_response = super().getresponse()
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/http/client.py", line 1377, in getresponse
    response.begin()
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/http/client.py", line 320, in begin
    version, status, reason = self._read_status()
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/http/client.py", line 281, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/socket.py", line 704, in readinto
    return self._sock.recv_into(b)
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/ssl.py", line 1242, in recv_into
    return self.read(nbytes, buffer)
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/ssl.py", line 1100, in read
    return self._sslobj.read(len, buffer)
urllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetError(54, 'Connection reset by peer'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/alexmann/Documents/Code_Learning/nhl_fromscratch/player_api.py", line 260, in <module>
    fetch_and_store_player_data()
  File "/Users/alexmann/Documents/Code_Learning/nhl_fromscratch/player_api.py", line 194, in fetch_and_store_player_data
    player_id = get_player_id(home_team, away_team, player_name)
  File "/Users/alexmann/Documents/Code_Learning/nhl_fromscratch/player_api.py", line 55, in get_player_id
    team2_id = get_NHL_abbreviations(team2)
  File "/Users/alexmann/Documents/Code_Learning/nhl_fromscratch/player_api.py", line 137, in get_NHL_abbreviations
    response = requests.get(f"{base_url}/standings/now")
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/requests/api.py", line 73, in get
    return request("get", url, params=params, **kwargs)
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/requests/sessions.py", line 724, in send
    history = [resp for resp in gen]
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/requests/sessions.py", line 724, in <listcomp>
    history = [resp for resp in gen]
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/requests/sessions.py", line 265, in resolve_redirects
    resp = self.send(
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "/Users/alexmann/.pyenv/versions/3.9.16/lib/python3.9/site-packages/requests/adapters.py", line 682, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', ConnectionResetError(54, 'Connection reset by peer'))
